src/twisted/web/client.py

```diff
@@ -411,14 +411,21 @@ def _cancelTimeout(self, result, timeoutCall):
         return result
 
     def gotHeaders(self, headers):
+        """
+        Parse the response HTTP headers.
+
+        @param headers: The response HTTP headers.
+        @type headers: L{list}
```

Comment:
`headers` cannot possibly be a `list`: it has to be string-indexable. The rest of the code suggests it's a `dict`.